addOns/domxss/src/main/java/org/zaproxy/zap/extension/domxss/DomXssScanRule.java

```diff
@@ -249,12 +250,12 @@ private WebDriver createWebDriver() {
     private WebDriverWrapper getWebDriver() {
         WebDriverWrapper fxDriver;
         try {
-            fxDriver = freeFirefoxDrivers.pop();
+            fxDriver = freeDrivers.pop();
```

Comment:
This needs to take into account the browser set, otherwise it might return a previous browser (e.g. if the user later changed to Chrome).